cmd/common_konnect.go

```diff
@@ -28,6 +28,11 @@ func syncKonnect(ctx context.Context,
 		return err
 	}
 
+	err = targetContent.PopulateDocumentContent(filenames)
```

Comment:
Question for my understanding: Why is this a separate function and not not done when the content is being read/built?

Reply:
The preceding function, `file.GetContentFromFiles()`, is a basic validation check before we call `getContent()`. `getContent()` iterates over all files and and [calls `readContent()` on each](https://github.com/Kong/deck/blob/v1.6.0/file/readfile.go#L30), which is in turn basically just a [passthrough to a YAML unmarshaler](https://github.com/Kong/deck/blob/v1.6.0/file/readfile.go#L94).

We can't handle reading Document content within the YAML unmarshaler, so we need some sort of separate function for that. We could place it at the end of `getContent()`, but this seems a better home for it as it's Konnect-specific, and `getContent()` runs for everything.